File: HandSock-Server/src/main/java/com/server/handsock/clients/service/ClientChatService.java
package com.server.handsock.clients.service;

import org.springframework.stereotype.Service;
import com.server.handsock.console.ConsolePrints;
import com.server.handsock.console.HandleResults;
import com.server.handsock.clients.dao.ClientChatDao;
import com.server.handsock.clients.mod.ClientChatModel;
import com.server.handsock.clients.man.ClientChatManage;
import org.springframework.beans.factory.annotation.Autowired;
import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper;

import java.util.Map;

@Service
public class ClientChatService {

    private final ClientChatDao clientChatDao;
    private final HandleResults handleResults = new HandleResults();

    @Autowired
    public ClientChatService(ClientChatDao clientChatDao) {
        this.clientChatDao = clientChatDao;
    }

    public ClientChatModel searchHistoryById(String sid) {
        try {
            return clientChatDao.selectOne(new QueryWrapper<ClientChatModel>().eq("sid", sid));
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
            new HandleResults().handleResultByCode(500, null, "服务端异常");
        }
        return null;
    }

    public Map<String, Object> insertChatMessage(String type, long uid, long gid, String address, String content) {
        try {
            ClientChatModel OPChatModel = new ClientChatModel();
            Map<String, Object> result = new ClientChatManage().insertChatMessage(OPChatModel, type, uid, gid, address, content);
            if (clientChatDao.insert(OPChatModel) > 0) {
                return handleResults.handleResultByCode(200, result, "发送成功");
            } else return handleResults.handleResultByCode(400, null, "发送失败");
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
            return handleResults.handleResultByCode(500, null, "服务端异常");
        }
    }

    public Map<String, Object> searchAllChatHistory(Long gid) {
        try {
            QueryWrapper<ClientChatModel> wrapper = new QueryWrapper<>();
            wrapper.orderByAsc("time");
            return handleResults.handleResultByCode(200, clientChatDao.selectList(wrapper.eq("gid", gid)), "获取成功");
        } catch (Exception e) {
            new ConsolePrints().printErrorLog(e);
            return handleResults.handleResultByCode(500, null, "服务端异常");
        }
    }
}

File: HandSock-Server/src/main/java/com/server/handsock/clients/service/ClientChatService.kt
package com.server.handsock.clients.service

import com.baomidou.mybatisplus.core.conditions.query.QueryWrapper
import com.server.handsock.clients.dao.ClientChatDao
import com.server.handsock.clients.man.ClientChatManage
import com.server.handsock.clients.mod.ClientChatModel
import com.server.handsock.utils.ConsoleUtils
import com.server.handsock.utils.HandUtils
import com.server.handsock.utils.IDGenerator
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service

@Service
class ClientChatService @Autowired constructor(private val clientChatDao: ClientChatDao) {
    fun insertChatMessage(type: String, uid: Long, gid: Long, address: String, content: String): Map<String, Any> {
        try {
            if (content.length > 200) return HandUtils.handleResultByCode(400, null, "消息过长")
            val clientChatModel = ClientChatModel()
            val result = ClientChatManage(HandUtils, ConsoleUtils, IDGenerator).insertChatMessage(clientChatModel, type, uid, gid, address, content)
            return if (clientChatDao.insert(clientChatModel) > 0) {
                HandUtils.handleResultByCode(200, result, "发送成功")
            } else HandUtils.handleResultByCode(400, null, "发送失败")
        } catch (e: Exception) {
            return HandUtils.printErrorLog(e)
        }
    }

    fun searchAllChatHistory(gid: Long?): Map<String, Any> {
        try {
            val wrapper = QueryWrapper<ClientChatModel>()
            wrapper.orderByAsc("time")
            return HandUtils.handleResultByCode(200, clientChatDao.selectList(wrapper.eq("gid", gid)), "获取成功")
        } catch (e: Exception) {
            return HandUtils.printErrorLog(e)
        }
    }
}
